Userland/Libraries/LibSyntax/Highlighter.cpp

```diff
@@ -164,4 +164,30 @@ StringView Highlighter::language_string(Language language) const
     }
 }
 
+StringView Highlighter::language_extension(Language language) const
```

Comment:
Seems weird for this to be a instance function, instead of a free function or at least a static member function. Or to be an instance function, it would make much more sense to just not have a language parameter and instead switch on `language()` below.

It's also missing the PlainText case. The `VERIFY_NOT_REACHED` call could be moved outside the switch statement to the bottom of the function (and the default case can be removed). That'll give you a compile error for the missing case, and more importantly, will give an error to anyone adding a new syntax in the future but forgets to add a case here.

Reply:
thanks for this quite detailed comment. just force-pushed the branch with changes addressing your comment. :^)